File: glsl_types.py
def GLSL_type(a):
	if(isinstance(a, str)):
		return "int["+str(len(a))+"]"
	elif(isinstance(a, list)):
		return GLSL_type(a)+"["+str(len(a))+"]"
	elif(type(a) == bool):
		return "bool"

File: glsl_types.php
function glsl_type($a){
	if(is_bool($a)){
		return "bool";
	}
	if(is_float($a)){
		return "float";
	}
	else if(is_array($a)){
		return glsl_type($a) . "[" . count($a) . "]";
	}
}
